src/Symfony/Component/HttpKernel/EventListener/AbstractSessionListener.php

```diff
@@ -109,6 +133,54 @@ public function onKernelResponse(ResponseEvent $event)
              * it is saved will just restart it.
              */
             $session->save();
+
+            /*
+             * For supporting sessions in php runtime with runners like roadrunner or swoole the session
+             * cookie need to be written on the response object and should not be written by PHP itself.
+             */
+            $sessionName = $session->getName();
+            $sessionId = $session->getId();
+            $sessionCookiePath = $this->sessionOptions['cookie_path'] ?? '/';
+            $sessionCookieDomain = $this->sessionOptions['cookie_domain'] ?? null;
+            $sessionCookieSecure = $this->sessionOptions['cookie_secure'] ?? false;
+            $sessionCookieHttpOnly = $this->sessionOptions['cookie_httponly'] ?? true;
+            $sessionCookieSameSite = $this->sessionOptions['cookie_samesite'] ?? Cookie::SAMESITE_LAX;
+
+            SessionUtils::popSessionCookie($sessionName, $sessionCookiePath);
+
+            $request = $event->getRequest();
+            $requestSessionCookieId = $request->cookies->get($sessionName);
+
+            if ($requestSessionCookieId && $session->isEmpty()) {
+                $response->headers->clearCookie(
+                    $sessionName,
+                    $sessionCookiePath,
+                    $sessionCookieDomain,
+                    $sessionCookieSecure,
+                    $sessionCookieHttpOnly,
+                    $sessionCookieSameSite
+                );
+            } elseif ($sessionId !== $requestSessionCookieId) {
```

Comment:
I have one silly question. Im asking because I'm ignorant. 

The if-statement on line 154 checks "if there is an existing cookie and the session is empty". 

This if-statement checks "if we have a new session".
What if we updated a value in the existing session, then we nee to store it in a cookie too, right? Or am I missing something?

Reply:
So, I found the answer to my own question. 

It is only the sessionId that needs to be stored in the cookie. The actual data is stored on file (default for PHP), in Redis or whatever session handler you have. 

So there is only need to add or remove the cookie, which is what this code does.